### PR TITLE
Writes back openlibrary_work and openlibrary_edition to IA after import

### PR DESCRIPTION
**Pull request by mekarpeles**

### CODE REVIEW

File: openlibrary/catalog/add_book/__init__.py [hunk 1]
```diff
@@ -630,6 +630,8 @@ def load(rec):
 
         web.ctx.site.save_many(edits, 'import new book')
 
-    # update_ia_metadata_for_ol_edition(reply['edition']['key'].split('/')[2])
+    # Writes back `openlibrary_edition` and `openlibrary_work` to
+    # archive.org item after successful import:
+    update_ia_metadata_for_ol_edition(reply['edition']['key'].split('/')[2])
```
Comment (bfalling) [inline openlibrary/catalog/add_book/__init__.py]: Will `reply['edition']` always yield a valid dictionary? Will the result of `split()` always have at least 3 elements? i.e. Is there some defensive protection needed here?

**Comment by mekarpeles** [comment]
thank you! good feedback. I think it does pay to have a check here -- will update

File: openlibrary/plugins/importapi/code.py [hunk 2]
```diff
@@ -181,7 +181,10 @@ def POST(self):
         if not can_write():
             return json.dumps({'success':False, 'error':'Permission Denied'})
 
-        i = web.input()
+        i = web.input()        
+
+        require_marc = False if i.get('require_marc') == 'false' else True
```
Comment (bfalling) [inline openlibrary/plugins/importapi/code.py]: Would this be more concise as `require_marc = not (i.get('require_marc') == 'false')`?

File: openlibrary/plugins/importapi/code.py [hunk 3]
```diff
@@ -224,26 +227,36 @@ def POST(self):
...
-            return self.error("item-is-serial")
-
-        # insider note: follows Archive.org's approach of
-        # Item::isMARCXMLforMonograph() which excludes non-books
-        if not (marc_leaders[7] == 'm' and marc_leaders[6] == 'a'):
-            return self.error("item-not-book")
+        else:
+            edition_data = self.get_ia_record(metadata)
+            print("!" * 100)
+            print(edition_data)
```
Comment (bfalling) [inline openlibrary/plugins/importapi/code.py]: Should these debug lines be left in here?

File: scripts/manage-imports.py [hunk 4]
```diff
@@ -71,6 +71,10 @@ def import_ocaids(*ocaids, **kwargs):
                 import-all
     """
     servername = kwargs.get('servername', None)
+    require_marc = not kwargs.get('no_marc', False)
+
+    print("require_marc?!?!?!: %s" % require_marc)
```
Comment (bfalling) [inline scripts/manage-imports.py]: Remove debug line?

File: scripts/manage-imports.py [hunk 5]
```diff
@@ -167,27 +173,31 @@ def main():
     load_config(configfile)
 
     from infogami import config
-    servername = config.get('servername', 'https://openlibrary.org')
 
     cmd = sys.argv[1]
-    args = sys.argv[2:]
+    args, flags = [], {'servername': config.get('servername', 'https://openlibrary.org')}
```
Comment (bfalling) [inline scripts/manage-imports.py]: Should this default servername be grabbed from or related to values in olsystem?

File: scripts/manage-imports.py [hunk 6]
```diff
@@ -167,27 +173,31 @@ def main():
...
+    for i in  sys.argv[2:]:
```
Comment (bfalling) [inline scripts/manage-imports.py]: Extra space after `in`.

File: openlibrary/plugins/importapi/code.py (hunk 2 above)
Comment (mekarpeles) [reply openlibrary/plugins/importapi/code.py]: great suggestion

**Pull request merged**